Add Distance Measurement submodule and sample (#6428)
nrfconnect/sdk-nrf: samples/bluetooth/nrf_dm/README.rst

.. _ble_nrf_dm:

Bluetooth: nRF Distance Measurement with Bluetooth LE discovery
###############################################################

.. contents::
   :local:
   :depth: 2

The nRF Distance Measurement sample demonstrates the functionality of the Distance Measurement (DM) subsystem.
It shows how to use DM to measure the distance between devices.

> The Bluetooth Heart Rate Service is running simultaneously.

Reviewer: ```suggestion
The Bluetooth® Heart Rate Service is running simultaneously.
```
The trademark sign is only needed for the first instance of Bluetooth.

> 

Reviewer: Move the Requirements section here to align with the sample template: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/templates/sample_README.html

Overview
********

The sample initializes and performs distance measurements between devices.
The procedure for distance measurement on both devices must be synchronized.

The synchronisation step has two goals:

   * Detect peers that support the ranging procedure.
   * Provide synchronisation for the ranging step.

> This application synchronizes devices using the scan and advertising facilities of the Bluetooth LE.

Reviewer: ```suggestion
This sample synchronizes devices using the scan and advertising facilities of the Bluetooth LE.
```

Anchoring occurs after the exchange of ``SCAN_REQ`` and ``SCAN_RSP`` packets.

.. figure:: images/sync.svg
   :alt: Synchronisation process

   Synchronisation process

> Both devices act as advertiser and scanner.

Reviewer: ```suggestion
Both devices act as an advertiser and a scanner.
```

> Device B receives an advertising from Device A and scans Device A.

Reviewer: ```suggestion
Device B receives advertising from Device A and scans it.
```

> Device A sends a scan response which contains ``Manufacturer Specific Data``.

Reviewer: ```suggestion
Device A sends a scan response that contains ``Manufacturer Specific Data``.
```

This message contains the following fields:

   * Company Identifier Code.
   * Identifier indicating that the device supports distance measurement.
   * The access address used for measurement.

This scanner/advertiser interaction is used as a synchronisation point.
After a configurable delay, Device A and Device B try to range each other.

>    * If a Device has acted as an advertiser in synchronization, it will act as the reflector in the ranging procedure.
>    * If a Device has acted as a scanner in synchronization, it will act as the initiator in the ranging procedure.

Reviewer: ```suggestion
   * If a Device acts as an advertiser in synchronization, it will act as the reflector in the ranging procedure.
   * If a Device acts as a scanner in synchronization, it will act as the initiator in the ranging procedure.
```

The synchronization will be performed cyclically.

Based on the ``SCAN_RSP`` response, the device recognizes peers that support the distance measurement functionality within its coverage.
The addresses of these devices are stored in a list.
When a device does not respond within a certain period of time, it is removed from the list.

> This list also stores the last measurement results with the peers.

Reviewer: ```suggestion
Also the last measurement results with the peers are stored in this list.
```

Requirements
************

The sample supports the following development kits:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf52840dk_nrf52840, nrf52833dk_nrf52833

User interface
**************

LED 1:
   Indicates the measured distance for MCPD (Multi-carrier phase difference) method. The shorter the distance, the brighter the light.

LED 2:
   Blinks every two seconds with the duty cycle set to 50% when the main loop is running and the device is advertising.

LED 3:
   On when connected.

Building and running
********************
.. |sample path| replace:: :file:`samples/bluetooth/nrf_dm`

.. include:: /includes/build_and_run.txt

Testing
=======

After programming the sample to development kits, test it by performing the following steps:

1. |connect_terminal_specific|
#. Reset the kit.
#. Program the other development kit with the same sample and reset it.
#. Wait until the devices identify themselves and synchronize.
#. Observe that the received measurement results are output in the terminal window.

Sample output
=============

> The result should look similar to the following output.

Reviewer: ```suggestion
The result should look similar to the following output:
```
